frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/state-update/vm-common-templates-updater.ts

```diff
@@ -0,0 +1,93 @@
+import {
+  hasVMSettingsValueChanged,
+  iGetRelevantTemplateSelectors,
+} from '../../selectors/immutable/vm-settings';
+import { CloudInitField, VMSettingsField, VMWizardProps, VMWizardStorageType } from '../../types';
+import { InternalActionType, UpdateOptions } from '../types';
+import { iGetLoadedCommonData } from '../../selectors/immutable/selectors';
+import {
+  iGetCommonTemplateCloudInit,
+  iGetRelevantTemplate,
+} from '../../../../selectors/immutable/template/combined';
+import { CLOUDINIT_DISK, DiskType, DiskBus, VolumeType } from '../../../../constants/vm';
+import { vmWizardInternalActions } from '../internal-actions';
+import {
+  CloudInitDataFormKeys,
+  CloudInitDataHelper,
+  generateCloudInitPassword,
+} from '../../../../k8s/wrapper/vm/cloud-init-data-helper';
+import { iGet, iGetIn, toShallowJS } from '../../../../utils/immutable';
+import { DiskWrapper } from '../../../../k8s/wrapper/vm/disk-wrapper';
+import { VolumeWrapper } from '../../../../k8s/wrapper/vm/volume-wrapper';
+import { iGetStorages } from '../../selectors/immutable/storage';
+
+export const commonTemplatesUpdater = ({ id, prevState, dispatch, getState }: UpdateOptions) => {
+  const state = getState();
+  const iUserTemplate = iGetLoadedCommonData(state, id, VMWizardProps.userTemplate);
+
+  if (
+    !hasVMSettingsValueChanged(
+      prevState,
+      state,
+      id,
+      VMSettingsField.OPERATING_SYSTEM,
+      VMSettingsField.FLAVOR,
+      VMSettingsField.WORKLOAD_PROFILE,
+    ) ||
+    iUserTemplate
+  ) {
+    return;
+  }
+  const iCloudInitStorage = iGetStorages(state, id).find((stor) =>
+    iGetIn(stor, ['volume', 'cloudInitNoCloud']),
+  );
+
+  const relevantOptions = iGetRelevantTemplateSelectors(state, id);
+  const iCommonTemplates = iGetLoadedCommonData(state, id, VMWizardProps.commonTemplates);
+  const iTemplate = iCommonTemplates && iGetRelevantTemplate(iCommonTemplates, relevantOptions);
+  const cloudInitHelper = new CloudInitDataHelper(
+    toShallowJS(iGetCommonTemplateCloudInit(iTemplate)),
+  );
+  // generate new password if the common template has a password for cloud ionit
+  if (cloudInitHelper.get(CloudInitDataFormKeys.PASSWORD)) {
+    cloudInitHelper.set(CloudInitDataFormKeys.PASSWORD, generateCloudInitPassword());
+
+    dispatch(
+      vmWizardInternalActions[InternalActionType.SetCloudInitFieldValue](
+        id,
+        CloudInitField.IS_FORM,
+        false,
+      ),
+    );
+    dispatch(
+      vmWizardInternalActions[InternalActionType.UpdateStorage](id, {
+        id: iGet(iCloudInitStorage, 'id'),
+        type: iGet(iCloudInitStorage, 'type') || VMWizardStorageType.TEMPLATE_CLOUD_INIT,
+        disk: new DiskWrapper()
```

Comment:
you can use the disk and volume you already have and `setType` This way user changes in bus/etc will be preserved.

Reply:
but this change will only occur in the first step of the wizard, its only right that the disk and volume will reset.

Reply:
ok then, let's reset everything